File: .taira-source-c80bf4d85a97/java/norito_java/src/main/java/org/hyperledger/iroha/norito/NoritoDump.java
// Copyright 2024 Hyperledger Iroha Contributors
// SPDX-License-Identifier: Apache-2.0

package org.hyperledger.iroha.norito;

import java.nio.file.Files;
import java.nio.file.Path;

/** Simple CLI to print Norito header information. */
public final class NoritoDump {
  private NoritoDump() {}

  public static void main(String[] args) throws Exception {
    if (args.length != 1) {
      System.err.println("Usage: NoritoDump <path>");
      System.exit(1);
    }
    byte[] data = Files.readAllBytes(Path.of(args[0]));
    NoritoHeader.DecodeResult result = NoritoHeader.decode(data, null);
    NoritoHeader header = result.header();
    System.out.println("Norito header:");
    System.out.println("  version: " + NoritoHeader.MAJOR_VERSION + "." + NoritoHeader.MINOR_VERSION);
    System.out.println("  schema hash: " + bytesToHex(header.schemaHash()));
    System.out.println("  compression: " + header.compression());
    System.out.println("  payload length: " + header.payloadLength());
    System.out.printf("  checksum: 0x%016x%n", header.checksum());
    System.out.printf("  flags: 0x%02x%n", header.flags());
  }

  private static String bytesToHex(byte[] bytes) {
    StringBuilder sb = new StringBuilder();
    for (byte b : bytes) {
      sb.append(String.format("%02x", b));
    }
    return sb.toString();
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/norito/NoritoDump.kt
// Copyright 2024 Hyperledger Iroha Contributors
// SPDX-License-Identifier: Apache-2.0

package org.hyperledger.iroha.sdk.norito

import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths

/** Simple CLI to print Norito header information. */
object NoritoDump {

    @JvmStatic
    fun main(args: Array<String>) {
        if (args.size != 1) {
            System.err.println("Usage: NoritoDump <path>")
            System.exit(1)
        }
        val data = Files.readAllBytes(Paths.get(args[0]))
        val result = NoritoHeader.decode(data, null)
        val header = result.header
        println("Norito header:")
        println("  version: ${NoritoHeader.MAJOR_VERSION}.${NoritoHeader.MINOR_VERSION}")
        println("  schema hash: ${bytesToHex(header.schemaHash)}")
        println("  compression: ${header.compression}")
        println("  payload length: ${header.payloadLength}")
        println("  checksum: 0x${"%016x".format(header.checksum)}")
        println("  flags: 0x${"%02x".format(header.flags)}")
    }

    private fun bytesToHex(bytes: ByteArray): String =
        bytes.joinToString("") { "%02x".format(it) }
}
